charts/ingress-nginx/templates/_helpers.tpl

```diff
@@ -114,6 +114,17 @@ Return the appropriate apiVersion for deployment.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return the appropriate apiVersion for daemonset.
+*/}}
+{{- define "daemonset.apiVersion" -}}
+{{- if semverCompare ">=1.9-0" .Capabilities.KubeVersion.GitVersion -}}
```

Comment:
You don't think is time to raise the minimum k8s version? 
Not sure makes sense keep this compatibility for something is not supported in the main k/k repository. Just a thought.

Reply:
I'm for upping the supported k8s version for this chart. The purpose of this pr is just to standardize the chart to make it easier to read and fix some of the small inconsistencies.
Are we also going to define a minimum supported version of ingress-nginx?

Reply:
> Are we also going to define a minimum supported version of ingress-nginx?

I don't want to say yes, but in the last week, I've seen several issues to ingress-nginx 0.20.0 (almost 18 months old). 
Can we limit the tag to not be less than three versions old? If the user decides to use `helm template` and using something older than that, fine, but it will required additional effort.
Makes sense?

Reply:
> I'm for upping the supported k8s version for this chart.

What is the version now? (stable/nginx-ingress)

Reply:
We have been updating the chart to the latest version of ingress-nginx when it is released. As apart of the repo I'd like to have something within the release handling that just magically updates it :)

`kubeVersion: ">=1.10.0-0"` This is defined in the chart.yaml

Reply:
We could provide a requirement that to run the chart it has to be at least three versions behind. Then if someone wants to run a version older they would have to use an older version of the chart.

Reply:
Do you want to do  ^^ here or in a different PR?
(merging as-is :+1 to me)

Reply:
I'd prefer that in another pr. Let's add that to the issue for the chart.